lib/libzpool/zrepl_mgmt.c

```diff
@@ -27,6 +26,54 @@ struct zvol_list stale_zv_list;
 
 static int uzfs_zinfo_free(zvol_info_t *zinfo);
 
+enum zrepl_log_level zrepl_log_level;
+
+/*
+ * Log message to stdout/stderr if log level allows it.
+ */
+void
+zrepl_log(enum zrepl_log_level lvl, const char *fmt, ...)
+{
+	va_list args;
+	struct timeval tv;
+	struct tm *timeinfo;
+	unsigned int ms;
+	char line[512];
+	FILE *outf;
+
+	if (lvl < zrepl_log_level)
```

Comment:
let us set 'likely' for this

Reply:
good one 👍  we could also modify debug macro not to call the zrepl_log function if not at debug level to save one function call. But I think a couple more assembler instructions won't kill us considering speed of zfs :-)